internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go

```diff
@@ -107,25 +113,25 @@ func verifyPipelines(spec *loggingv1.ClusterLogForwarderSpec, status *loggingv1.
 	inputs := sets.StringKeySet(spec.InputMap())
 
 	for i, pipeline := range spec.Pipelines {
-		// Don't allow empty names since this no longer mutates the spec
+		// Give anonymous pipelines internal "pipeline_[0..n]" names
```

Comment:
Validation is no longer used to mutate the spec and this will further change once https://github.com/openshift/cluster-logging-operator/pull/2008 lands.  We should add a "migration" to add the name if it is not defined to resolve this